server/base/src/main/java/org/apache/accumulo/server/util/checkCommand/SystemConfigCheckRunner.java

```diff
@@ -18,22 +18,230 @@
  */
 package org.apache.accumulo.server.util.checkCommand;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.lock.ServiceLock;
+import org.apache.accumulo.core.metadata.AccumuloTable;
+import org.apache.accumulo.core.metadata.TServerInstance;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata;
+import org.apache.accumulo.core.zookeeper.ZooSession;
 import org.apache.accumulo.server.ServerContext;
 import org.apache.accumulo.server.cli.ServerUtilOpts;
+import org.apache.accumulo.server.log.WalStateManager;
 import org.apache.accumulo.server.util.Admin;
 
 public class SystemConfigCheckRunner implements CheckRunner {
   private static final Admin.CheckCommand.Check check = Admin.CheckCommand.Check.SYSTEM_CONFIG;
 
+  public enum ServerProcess {
+    MANAGER, GC, TSERVER, COMPACTION_COORDINATOR, COMPACTOR, MONITOR, SCAN_SERVER
+  }
+
   @Override
   public Admin.CheckCommand.CheckStatus runCheck(ServerContext context, ServerUtilOpts opts,
       boolean fixFiles) throws Exception {
     Admin.CheckCommand.CheckStatus status = Admin.CheckCommand.CheckStatus.OK;
     printRunning();
+
+    log.trace("********** Checking validity of some ZooKeeper nodes **********");
+    status = checkZkNodes(context, status);
+
     printCompleted(status);
     return status;
   }
 
+  private static Admin.CheckCommand.CheckStatus checkZkNodes(ServerContext context,
+      Admin.CheckCommand.CheckStatus status) throws Exception {
+    status = checkZKLocks(context, status);
+    status = checkZKTableNodes(context, status);
+    status = checkZKWALsMetadata(context, status);
+
+    return status;
+  }
+
+  private static Admin.CheckCommand.CheckStatus checkZKLocks(ServerContext context,
+      Admin.CheckCommand.CheckStatus status) throws Exception {
+    final ServerProcess[] serverProcesses = ServerProcess.values();
+    final String zkRoot = context.getZooKeeperRoot();
+    final var zs = context.getZooSession();
+    final var zrw = zs.asReaderWriter();
+    final var compactors = context.instanceOperations().getCompactors();
+    final var sservers = context.instanceOperations().getScanServers();
```

Comment:
Looking back at this code again: this gets the compactors and sservers from scanning ZK then validating the lock node... I hadn't considered how this was gathering the data. Thinking most or all of the code for the cases for COMPACTOR and SCAN_SERVER below can be removed. Thinking the new cases would look something like:
```
case COMPACTOR:
	var compactors = context.instanceOperations().getCompactors();
	if (compactors.isEmpty()) {
		// warn user (debug) that no compactors were found to be running which may or may not be expected.
	}
	break;
```